pallets/vc-management/src/lib.rs

```diff
@@ -165,5 +245,98 @@ pub mod pallet {
+
+		// Change the schema Admin account
+		#[pallet::weight(195_000_000)]
+		pub fn set_schema_admin(
+			origin: OriginFor<T>,
+			new: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			T::SetAdminOrigin::ensure_origin(origin)?;
+			Self::deposit_event(Event::SchemaAdminChanged {
+				old_admin: Self::schema_admin(),
+				new_admin: Some(new.clone()),
+			});
+			<SchemaAdmin<T>>::put(new);
+			Ok(Pays::No.into())
+		}
+
+		// It requires the schema Admin account
+		#[pallet::weight(195_000_000)]
+		pub fn add_schema(
+			origin: OriginFor<T>,
+			id: Vec<u8>,
+			content: Vec<u8>,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+			ensure!(Some(sender.clone()) == Self::schema_admin(), Error::<T>::RequireSchemaAdmin);
+
+			
+
+			let index = Self::schema_count();
+			<SchemaCount<T>>::put(index + 1);
+			SchemaRegistry::<T>::insert(
+				index,
+				VCSchema::<T>::new(id.clone(), sender.clone(), content.clone()),
+			);
+			Self::deposit_event(Event::SchemaIssued { account: sender, index });
+			Ok(().into())
+		}
+
+		// It requires the schema Admin account
+		#[pallet::weight(195_000_000)]
+		pub fn disable_schema(
+			origin: OriginFor<T>,
+			index: SchemaIndex,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+			ensure!(Some(sender.clone()) == Self::schema_admin(), Error::<T>::RequireSchemaAdmin);
+
+			SchemaRegistry::<T>::try_mutate(index, |context| {
+				let mut c = context.take().ok_or(Error::<T>::SchemaNotExists)?;
+				ensure!(c.status == Status::Active, Error::<T>::SchemaAlreadyDisabled);
+				c.status = Status::Disabled;
+				*context = Some(c);
+				Self::deposit_event(Event::SchemaDisabled { account: sender, index });
+				Ok(().into())
+			})
+		}
+
+		// It requires the schema Admin account
+		#[pallet::weight(195_000_000)]
+		pub fn active_schema(
+			origin: OriginFor<T>,
+			index: SchemaIndex,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+			ensure!(Some(sender.clone()) == Self::schema_admin(), Error::<T>::RequireSchemaAdmin);
+			SchemaRegistry::<T>::try_mutate(index, |context| {
+				let mut c = context.take().ok_or(Error::<T>::SchemaNotExists)?;
+				ensure!(c.status == Status::Disabled, Error::<T>::SchemaAlreadyActivated);
+				c.status = Status::Active;
+				*context = Some(c);
+				Self::deposit_event(Event::SchemaActivated { account: sender, index });
+				Ok(().into())
+			})
+		}
+
+		// It requires the schema Admin account
+		#[pallet::weight(195_000_000)]
+		pub fn revoke_schema(
+			origin: OriginFor<T>,
+			index: SchemaIndex,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+			ensure!(Some(sender.clone()) == Self::schema_admin(), Error::<T>::RequireSchemaAdmin);
+
+			let context = SchemaRegistry::<T>::get(index).ok_or(Error::<T>::SchemaNotExists)?;
+			SchemaRegistry::<T>::remove(index);
+			let index = Self::schema_count();
+			if index > 0 {
+				<SchemaCount<T>>::put(index - 1);
```

Comment:
I don't think this will work :( we don't always remove the last element in `SchemaRegistry`, as a result, we get correct SchemaCount but wrong index when adding a new schema. 

Imagine we have 4 schemas `[1, 4]`, we revoke schema idx = 2, and for the next `add_schema` call we would have idx = 4 which is already occupied.

A simple solution is to always increment the index and do not care if there are indexes revoked.

Reply:
get it, is fixing now